lib/uri_parser.js

```diff
@@ -270,6 +272,20 @@ function applyConnectionStringOption(obj, key, value, options) {
     }
   }
 
+  if (key === 'readpreferencetags' && Array.isArray(value)) {
+    const parsedTags = [];
+
+    for (let i = 0; i < value.length; i++) {
+      parsedTags[i] = {};
```

Comment:
this seems ripe for refactoring, can we pull this out into a well-named method and add a unit test or two stressing the implementation out?